packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -462,6 +462,65 @@ class LinuxPlugin extends PluginPlatform implements NativeOrDartPlugin {
   }
 }
 
+class CustomEmbedderPlugin extends PluginPlatform implements NativeOrDartPlugin {
+  const CustomEmbedderPlugin({
+    required this.embedderName,
+    required this.name,
+    required this.pluginPath,
+    required FileSystem fileSystem,
+    this.dartPluginClass,
+    this.defaultPackage,
+  }) : configKey = 'x-$embedderName',
+       _fileSystem = fileSystem;
+
+  factory CustomEmbedderPlugin.fromYaml({
+    required String embedderName,
+    required String name,
+    required YamlMap yaml,
+    required String pluginPath,
+    required FileSystem fileSystem
+  }) {
+    assert(validate(yaml));
```

Comment:
so asserts don't run in the tool unless you explicitly enable them by editing the shell script entrypoint. should we run this validation always?

Reply:
The check in https://github.com/flutter/flutter/blob/90fa1b3be0e678704b709d1dd85cba0723814faf/packages/flutter_tools/lib/src/plugins.dart#L280 should've already made sure the yaml is valid (except I forgot to add some validation there, but I did that now)